utility_analysis_new/tests/combiners_test.py

```diff
@@ -158,8 +158,106 @@ def test_add_accumulators_moments(self):
 
     def test_partition_selection_accumulator_created(self):
         utility_analysis_combiner = combiners.UtilityAnalysisCountCombiner(
-            _create_combiner_params(), is_public_partitions=False)
+            _create_combiner_params_for_count(), is_public_partitions=False)
         acc = utility_analysis_combiner.create_accumulator(((2, 3, 4), 4))
         self.assertIsNotNone(acc.partition_selection_accumulator)
         self.assertSequenceAlmostEqual(
             [1 / 4], acc.partition_selection_accumulator.probabilities)
+
+
+def _create_combiner_params_for_sum(
+        min, max) -> pipeline_dp.combiners.CombinerParams:
+    return pipeline_dp.combiners.CombinerParams(
+        pipeline_dp.budget_accounting.MechanismSpec(
+            mechanism_type=pipeline_dp.MechanismType.GAUSSIAN,
+            _eps=1,
+            _delta=0.00001),
+        pipeline_dp.AggregateParams(
+            max_partitions_contributed=1,
+            max_contributions_per_partition=2,
+            min_sum_per_partition=min,
+            max_sum_per_partition=max,
+            noise_kind=pipeline_dp.NoiseKind.GAUSSIAN,
+            metrics=[pipeline_dp.Metrics.SUM],
+        ))
+
+
+class UtilityAnalysisSumCombinerTest(parameterized.TestCase):
+
+    @parameterized.named_parameters(
+        dict(testcase_name='empty',
+             num_partitions=0,
+             contribution_values=(),
+             params=_create_combiner_params_for_sum(0, 0),
+             expected_metrics=combiners.SumUtilityAnalysisMetrics(
+                 sum=0,
+                 per_partition_contribution_error=0,
+                 expected_cross_partition_error=0,
+                 std_cross_partition_error=0,
+                 std_noise=7.46484375,
+                 noise_kind=pipeline_dp.NoiseKind.GAUSSIAN)),
+        dict(testcase_name='one_partition_zero_partition_error',
+             num_partitions=1,
+             contribution_values=(1.1, 2.2),
+             params=_create_combiner_params_for_sum(0, 3.4),
+             expected_metrics=combiners.SumUtilityAnalysisMetrics(
+                 sum=3.3000000000000003,
```

Comment:
@dvadym This is the inaccurate test 😢 🐼

Reply:
Comparing the floating point number is tricky. Only if a floating point number is binary rational (i.e. can be presented in binary system with the number of digits not more than float64, e.g. `3`, `0.5`, `1/256`) the operation is precise. The usual approach in tests not to rely on the exact answer of floating point (which can be different on different CPUs), but to make approximate comparison.

Please compare each floating point field with `assertAlmostEqual`. Here is an
[example](https://github.com/OpenMined/PipelineDP/blob/2940e2ba8825321c6c8c8b5a0f2f6dc290477d2f/tests/dp_computations_test.py#L57) in code of using `self.assertAlmostEqual`,

Reply:
This works, thank you for the suggestion! I have left it at default precision (7 digits). Each field of SumUtilityAnalysisMetrics is now compared with almost equal, except noise type 😄 PLMK if there should be modifications. Thank you!